src/main/webapp/scripts/forum-script.js

```diff
@@ -179,3 +211,37 @@ function collapseReplies(idHandler) {
   $('#' + elementId + ' .expand-button').css('display', 'inline-block');
   $('#' + elementId + ' .collapse-button').css('display', 'none');
 }
+
+/**
+ *  Onclick handler for a user searching to reload forum with search parameter
+ */
+function search() {
+  const search = $('#search-input').val();
+  $('#search-input').val('');
+  const id = -1;
+  const placeholder = $('#forum-placeholder');
+  expandForumWithSearch(placeholder, id, search);
+}
+
+/**
+ *  Returns whether a word of the search is contained with in the text
+ *
+ *  @param {String} text contents of a question
+ *  @param {String} search contents of the user search
+ *
+ *  @return {Boolean} whether there is a word of the search within the text
+ */
+function containsSearch(text, search) {
```

Comment:
(Don't do this now) In a future iteration, it could be cool to calculate a "match score" and then sort the results based on that as a kind of relevance. There's a bunch of ways to do that, but a naive way to do that might be to just use `words.length` (or even the total number of letters in all `words`) as a score.

Reply:
Yeah I have been looking into implementing the search engine again.  I will consider a match score for the next iteration.